auraed/src/signal_handlers.rs

```diff
@@ -0,0 +1,48 @@
+/* -------------------------------------------------------------------------- *\
+ *             Apache 2.0 License Copyright © 2022 The Aurae Authors          *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use tokio::signal::unix::SignalKind;
+
+/// Waits for a [SIGTERM] signal and...
+/// * Calls [Cells::broadcast_free]
+/// * Calls [Cells::broadcast_kill]
+/// ---
+/// Returns after processing the first received signal.
+pub async fn terminate() {
+    let mut stream = tokio::signal::unix::signal(SignalKind::terminate())
+        .expect("failed to listen for SIGTERM");
+
+    let _ = stream.recv().await;
+    let mut cells = crate::runtime::CELLS.lock().await;
```

Comment:
should we instead send this signal to the cell service, which can handle the broadcast and kill calls on our behalf?  this means exposing the cell service instance to the handler rather than the implementation detail of the cells cache.

Reply:
I think this will mean we put cells back on CellService (get rid of the global variable). Make a clone of CellService (cheap since it's only ARCs in the struct) and pass that into each signal handler that we make (or make a signal handler struct). 

Then have additional broadcast methods on CellService. 

Or do you mean making a rpc call? (I'm not sure how that would look)

Either way, I think we would be unnecessarily going through CellService which I look at as a hopefully slim/dumb-ish handler for the grpc api

Reply:
i was hoping we'd have some instance of the CellService we could get at from the signal handler.  is there no way to make _that_ the global thing?  or get a reference to it from tokio?

i didn't mean make an RPC from inside the binary, no.  definitely not from a signal handler!

Reply:
Got it. I'll do this

Reply:
\o/ no globals!